src/qml/filters/fisheye/ui.qml

```diff
@@ -0,0 +1,650 @@
+            fn_SetPlugin(v0_amount, v1_deFish, v2_type, v3_scaling, v4_manualScale, v5_interpolator, v6_aspectType, v7_manualAspect)
+            filter.savePreset(defaultParameters)
+        }
+
+        filter.blockSignals = false
+        fn_ListCamera() //fill comboboxes
+
+        fn_SetControls() //set UI values
+    }
+
+    ///////////////////
+    //main UI gid
+    GridLayout {
+        columns: 3
+        anchors.fill: parent
+        anchors.margins: 8
+
+        // Row 1 Preset
+        Label {
+            text: qsTr('Presets')
+            Layout.alignment: Qt.AlignRight
+        }
+        Shotcut.Preset {
+            id: preset
+            parameters: defaultParameters
+            onPresetSelected: fn_SetControls()
+        }
+        Row {width: 6} //blank, keep above within grid
+
+        //line seperator #1
+        Rectangle{Layout.columnSpan: 3; Layout.fillWidth: true; height: 1; color: activePalette.text; opacity: 0.3}
+
+        // Row 2 defish/fish
+        Label {
+            text: qsTr('Fisheye')
+            Shotcut.HoverTip { text: qsTr('Add or Remove fisheye effect') }
+            Layout.alignment: Qt.AlignRight
+        }
+        RowLayout {
+            ButtonGroup { id: fisheyeGroup }
+            RadioButton {
+                id: uID1_defisheye
+                text: qsTr('Remove')
+                ButtonGroup.group: fisheyeGroup
+                checked: (filter.get(idx1_deFish) == true) ? true:false
+                onCheckedChanged: if (checked) filter.set(idx1_deFish, true)
+            }
+            RadioButton {
+                id: uID1_fisheye
+                text: qsTr('Add')
+                ButtonGroup.group: fisheyeGroup
+                checked: (filter.get(idx1_deFish) == false) ? true:false
+                onCheckedChanged: if (checked) filter.set(idx1_deFish, false)
+            }
+        }
+        Shotcut.UndoButton {
+            onClicked: {
+                if(v1_deFish == true)
+                    uID1_defisheye.checked = true
+                else
+                    uID1_fisheye.checked = true
+            }
+        }
+
+        // Row 3: focal ratio
+        Label {
+            text: qsTr('Focal Ratio')
+            Shotcut.HoverTip { text: qsTr('The amount of lens distortion') }
+            Layout.alignment: Qt.AlignRight
+        }
+        Shotcut.SliderSpinner {
+            id: uID0_ratioSlider
+            minimumValue: 0
+            maximumValue: 1
+            decimals: 3
+            suffix: ''
+            value: filter.getDouble(idx0_amount)
+            onValueChanged: filter.set(idx0_amount, value)
+        }
+        Shotcut.UndoButton {
+            onClicked: uID0_ratioSlider.value = v0_amount
+        }
+
+        // Row 4: Interpolation Quality
+        Label {
+            text: qsTr('Quality')
+            Shotcut.HoverTip { text: qsTr('Resample quality\n 1=no interpolation\n 2=fast linear\n 7=slowest') }

```

Comment:
When you have to enumerate values in a tooltip that means this field should be a combo and not a numeric input.